### PR TITLE
feat: [CCIE-2375] ephemeral volume support

### PR DESCRIPTION
**Pull request by rzlim08**
<details open>
  <summary><a href="https://czi-tech.atlassian.net/browse/CCIE-2375" title="CCIE-2375" target="_blank">CCIE-2375</a></summary>
  <br />
  <table>
    <tr>
      <th>Summary</th>
  <td>Create shared empty volume mounts between containers</td>
    </tr>
    <tr>
      <th>Type</th>
      <td>
        Task
        Task
      </td>
    </tr>
    <tr>
      <th>Status</th>
      <td>Done</td>
    </tr>
  </table>
</details>

---

* builds off @jgadling's branch [here](https://github.com/chanzuckerberg/happy/tree/jgadling/ephemeral-volume-support)
* Creates empty volume mount
* Adds volume mount to main container, init container, and sidecar container
* [CCIE-2375](https://czi-tech.atlassian.net/browse/CCIE-2375)

# Description
copilot:summary

[CCIE-2375]: https://czi-tech.atlassian.net/browse/CCIE-2375?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: terraform/modules/happy-service-eks/main.tf [hunk 1]
```diff
@@ -535,14 +558,24 @@ resource "kubernetes_deployment_v1" "deployment" {
         }
 
         dynamic "volume" {
-          for_each = toset(var.additional_volumes_from_config_maps.items)
+          for_each = toset(var.additional_volumes_from_secrets.items)
```
Comment (rzlim08) [inline terraform/modules/happy-service-eks/main.tf:561]: @jgadling should this have changed?

Comment (jgadling) [reply terraform/modules/happy-service-eks/main.tf:561]: no... I'm not sure how that happened?

File: terraform/modules/happy-service-eks/main.tf [hunk 2]
```diff
@@ -458,7 +474,14 @@ resource "kubernetes_deployment_v1" "deployment" {
                 read_only  = true
               }
             }
-
+            dynamic "volume_mount" {
+              for_each = toset(var.emptydir_volumes)
+              content {
+                # TODO FIXME do we want the mount path to be configurable????
```
Comment (jgadling) [inline terraform/modules/happy-service-eks/main.tf:480]: @rzlim08 do you think we should add support for specifying mount paths, or delete these comments?

Comment (rzlim08) [reply terraform/modules/happy-service-eks/main.tf:480]: IMO there's no immediate need for it and a slight potential for the footgun of mounting over necessary folders, so I'll delete the comments. It'd be relatively easy to change in the future though.

**Review by jgadling — APPROVED** [review]

**Review by jakeyheath — APPROVED** [review]

**Pull request merged**